### PR TITLE
fix: return JSON errors for missing payment type params

### LINKED ISSUE #2857: [Bug] CreatePaymentType2Action / UpdatePaymentType2Action return an empty 200 (no JSON body) when required params are missing — client JS gets no error (claude assist)
**Type:** Bug
**Priority:** 4-Low
**Kind:** Task

## Summary

The payment-type AJAX mutators return a bare `null` Struts result both on the success path (after writing JSON) and on the missing-parameter guard path. On the guard path nothing is written to the response, so the client receives an empty HTTP 200 with no `{ret, reason}` body. The `editBillingPaymentType.jsp` AJAX handlers branch on `ret`, so a malformed/empty request silently looks like neither success nor failure.

## Locations

- `src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/CreatePaymentType2Action.java:83` (early `return null;` with no response written) and `:109`
- `src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/UpdatePaymentType2Action.java:86` and `:115`

## Suggested fix

1. On the validation-failure path, write the same JSON error envelope the success path uses (`{"ret":"1","reason":"..."}`) before returning.
2. For direct-response actions that have already written the body, return `NONE` rather than bare `null` so the dispatcher doesn't attempt to resolve a (missing) result — consistent with the repo convention that bare-null direct responses are unreliable.

This is presentation/robustness only; no privilege or data-exposure impact (the actions are `_billing` w gated and POST/CSRF protected).

🤖 Found during the Ontario Billing (OHIP) security audit.

### PR DESCRIPTION
**Pull request by nightcityblade**
## Description

Return a JSON error envelope when the create/update payment type actions are called without required parameters, and return `NONE` after writing direct JSON responses so Struts does not resolve a missing result.

## Related Issues

Fixes #2857

## How Was This Tested?

- `mvn -q -Dtest=CreatePaymentType2ActionUnitTest,UpdatePaymentType2ActionUnitTest test`

## Screenshots

N/A — backend JSON response handling only.

## Checklist

- [x] My commits are signed off for the [DCO](https://developercertificate.org/) (`git commit -s`)
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/) format, or I've written clear commit messages and will use the format next time
- [x] I have not included any patient data (PHI) in this PR
- [x] I have added tests for new functionality, or this change doesn't need new tests
- [x] I have read the [contributing guide](CONTRIBUTING.md)

## Summary by Sourcery

Return JSON error responses and Struts NONE results when payment type parameters are missing in create/update actions.

Bug Fixes:
- Ensure create payment type action returns a JSON error envelope and NONE when the paymentType parameter is missing.
- Ensure update payment type action returns a JSON error envelope and NONE when required payment type parameters are missing.
- Update actions to consistently return NONE instead of null after writing JSON responses to avoid Struts resolving a missing result.

Tests:
- Adjust create/update payment type unit tests to assert JSON error bodies and NONE results when parameters are missing.
- Update create/update payment type success-path tests to expect NONE instead of null as the execute() return value.

---
## Summary by cubic
Return consistent JSON errors when required payment type params are missing or blank, and always return Struts `NONE` after writing the response. Fixes #2857.

- **Bug Fixes**
  - `CreatePaymentType2Action`/`UpdatePaymentType2Action`: when `paymentType` (or `oldPaymentType`) is missing or blank, respond with `{"ret":"1","reason":"Missing payment type."}`.
  - Return `NONE` instead of `null` after any JSON write to prevent Struts result resolution.
  - Updated tests to assert `NONE` and the error JSON for missing/blank params; success-path tests expect `NONE`.

<sup>Written for commit a5ccca878338ba7e612e1b17035beac1233a29b1. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/carlos-emr/carlos/pull/2870?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/UpdatePaymentType2Action.java [hunk 1]
```diff
@@ -81,12 +81,15 @@ public String execute() {
...
         String paymentType = request.getParameter("paymentType");
+        Map<String, String> ret = new HashMap<>();
         if (oldPaymentType == null || oldPaymentType.isEmpty()
                 || paymentType == null || paymentType.isEmpty()) {
-            return null;
+            ret.put("ret", "1");
+            ret.put("reason", "Missing payment type.");
+            writeJsonResponse(response, ret);
+            return NONE;
         }
```
[the remark this answers is not in the document]
Comment (nightcityblade) [reply src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/UpdatePaymentType2Action.java]: I am keeping the shared error text intentionally. Issue 2857 asks for the same JSON error envelope used by the existing path, and the client contract branches on ret; changing the reason strings is optional and outside this focused fix.

File: src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/CreatePaymentType2Action.java [hunk 2]
```diff
@@ -79,11 +79,14 @@ public String execute() {
         }
 
         String paymentType = request.getParameter("paymentType");
+        Map<String, String> ret = new HashMap<>();
         if (paymentType == null || paymentType.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (nightcityblade) [reply src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/CreatePaymentType2Action.java]: Addressed in 9ca6c9bf. Create now uses String.isBlank(), with a focused regression test for whitespace-only input. The two focused action test classes pass (16 tests), and Checkstyle reports 0 violations.

File: src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/UpdatePaymentType2Action.java [hunk 3]
```diff
@@ -81,12 +81,15 @@ public String execute() {
 
         String oldPaymentType = request.getParameter("oldPaymentType");
         String paymentType = request.getParameter("paymentType");
+        Map<String, String> ret = new HashMap<>();
         if (oldPaymentType == null || oldPaymentType.isEmpty()
                 || paymentType == null || paymentType.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (nightcityblade) [reply src/main/java/io/github/carlos_emr/carlos/billings/ca/on/web/UpdatePaymentType2Action.java]: Addressed in 9ca6c9bf. Update now rejects blank old and new payment type values, with parameterized coverage for both cases. The two focused action test classes pass (16 tests), and Checkstyle reports 0 violations.

**Comment by Ben-Heerema** [comment]
@nightcityblade Once again sorry I didn't see this again until now. The fix looks great though, thank you!

**Comment by Ben-Heerema** [comment]
@yingbull ready for review

**Comment by nightcityblade** [comment]
Thanks for taking another look and moving this to review — much appreciated.

**Review by yingbull — APPROVED** [review]

**Comment by yingbull** [comment]
@nightcityblade we are back to merging and appreciate your help.
